cloudshell/cli/service/action_map.py

```diff
@@ -0,0 +1,192 @@
+from collections import OrderedDict
+import re
+
+from cloudshell.cli.session.session_exceptions import SessionLoopDetectorException
+
+
+class Action(object):
+    def __init__(self, pattern, callback, execute_once=False):
+        """
+
+        :param str pattern:
+        :param function callback:
+        :param bool execute_once:
+        """
+        self.pattern = pattern
+        self.callback = callback
+        self.execute_once = execute_once
+
+    def __call__(self, session, logger):
+        """
+
+        :param cloudshell.cli.session.expect_session.ExpectSession session:
+        :param logging.Logger logger:
+        :return:
+        """
+        return self.callback(session, logger)
+
+    def __repr__(self):
+        """
+
+        :rtype: str
+        """
+        return "{} pattern: {}, execute once: {}".format(super(Action, self).__repr__(),
```

Comment:
in python3 you can use super without args for getting base class `super().__repr__()`

Reply:
also in python3.6+ you can use `f"{self.pattern}"` which equalient to `"{}".format(self.pattern)`